test_runner/README.md

```diff
@@ -62,36 +62,64 @@ Exit after the first test failure:
 `pytest -x ...`
 (there are many more pytest options; run `pytest -h` to see them.)
 
+### Writing a test
 
-### Building new tests
+Every test needs a Zenith Environment, or ZenithEnv to operate in. A Zenith Environment
+is like a little cloud-in-a-box, and consists of a Pageserver, 0-N Safekeepers, and
+compute Postgres nodes. The connections between them can be configured to use JWT
+authentication tokens, and some other configuration options can be tweaked too.
 
-The tests make heavy use of pytest fixtures. You can read about how they work here: https://docs.pytest.org/en/stable/fixture.html
+The easiest way to get access to a Zenith Environment is by using the 'zenith_simple_env'
+fixture. The 'simple' env may be shared across multiple tests, so don't shut down the nodes
+or make other destructive changes in that environment.
 
-Essentially, this means that each time you see a fixture named as an input parameter, the function with that name will be run and passed as a parameter to the function.
+For more complicated cases, you can build a custom Zenith Environment, with the 'zenith_env'
+fixture:
 
-So this code:
-
-```python
-def test_something(zenith_cli, pg_bin):
-    pass
+```
+test_foobar(zenith_env: ZenithEnv):
+    # Prescribe the environment.
+	# We want to have 3 safekeeper nodes, and use JWT authentication in the
+	# connections to the page server
+	zenith_env.num_safekeepers = 3
+    zenith_env.set_pageserver_auth(True)
+
+    # Now create the environment. This initializes the repository, and starts
+	# up the page server and the safekeepers
+	env = zenith_env.init()
+
+    # Run the test
+	...
 ```
 
-... will run the fixtures called `zenith_cli` and `pg_bin` and deliver those results to the test function.
+For more information about pytest fixtures, see https://docs.pytest.org/en/stable/fixture.html
 
-Fixtures can't be imported using the normal python syntax. Instead, use this:
+ZenithEnv contains functions for stopping/starting nodes in the environment, checking
```

Comment:
I think text below should actually be a part of `ZenithEnv`'s docstring so it's closer to actual implementation. Similarly, I feel more like documenting `zenith_cli()` in method's docstring to avoid duplication.